server/service/src/requisition_line/response_line_stats/response_line_stats.rs

```diff
@@ -0,0 +1,113 @@
+use repository::{
+    EqualFilter, InvoiceLineFilter, InvoiceLineRepository, InvoiceLineRowType, InvoiceRowStatus,
+    InvoiceRowType, RepositoryError, RequisitionLine, RequisitionLineFilter,
+    RequisitionLineRepository, RequisitionRowStatus, RequisitionRowType, StockLineFilter,
+    StockLineRepository, StorageConnection,
+};
+
+#[derive(Clone, Debug, PartialEq, Default)]
+pub struct RequestStoreStats {
+    pub stock_on_hand: i32,
+    pub amc: i32,
+    pub max_months_of_stock: f64,
+    pub suggested_quantity: i32,
+}
+
+pub fn customer_store_stats(
+    requisition_line: &RequisitionLine,
+) -> Result<RequestStoreStats, RepositoryError> {
+    Ok(RequestStoreStats {
+        stock_on_hand: requisition_line
+            .requisition_line_row
+            .available_stock_on_hand,
+        amc: requisition_line
+            .requisition_line_row
+            .average_monthly_consumption,
+        max_months_of_stock: requisition_line.requisition_row.max_months_of_stock,
+        suggested_quantity: requisition_line.requisition_line_row.suggested_quantity,
+    })
+}
+
+#[derive(Clone, Debug, PartialEq, Default)]
+pub struct ResponseStoreStats {
+    pub stock_on_hand: f64,
+    pub stock_on_order: i32, // Internal Order
+    pub incoming_stock: i32, // Linked Inbound - Shipped
+    pub requested_quantity: i32,
+    pub other_requested_quantity: i32,
+}
+
+pub fn response_store_stats(
+    connection: &StorageConnection,
+    store_id: &str,
+    requisition_line: &RequisitionLine,
+) -> Result<ResponseStoreStats, RepositoryError> {
+    let stock_lines = StockLineRepository::new(connection).query_by_filter(
+        StockLineFilter::new()
+            .item_id(EqualFilter::equal_to(
+                &requisition_line.requisition_line_row.item_id,
+            ))
+            .store_id(EqualFilter::equal_to(&store_id)),
+    )?;
+
+    let stock_on_hand = stock_lines.iter().fold(0.0, |sum, stock_line| {
```

Comment:
soh should be in units I think - so number of packs * pack size

Reply:
[db8519f](https://github.com/openmsupply/open-msupply/pull/840/commits/db8519f03ceecbae9b69e82a68f8ed7dc5d1920c)